### ISSUE TITLE
[Impeller] TRect::Area can overflow

### ISSUE THREAD

**Issue by flar** [flutter/flutter#141710]
The `TRect::Area()` method currently uses a simply multiply to return its result which can easily overflow integer math dimensions. It should use a saturated operation on integers to return MAX_INT (or similar) when the area overflows the size of the integer coordinate type.

**Comment by jonahwilliams** [comment]
@flar was this fixed with your patch to migrate to the new Rect implementation?

**Comment by flar** [comment]
> @flar was this fixed with your patch to migrate to the new Rect implementation?

No, it was discovered at that time but seemed beyond the scope of that PR.
